src/transformers/modeling_flax_utils.py

```diff
@@ -169,31 +207,115 @@ def from_pretrained(cls, pretrained_model_name_or_path, dtype: jnp.dtype = jnp.f
         # Instantiate model.
         with open(resolved_archive_file, "rb") as state_f:
             try:
-                from flax.serialization import from_bytes
-
-                state = from_bytes(cls.model_class, state_f)
-            except TypeError:
-                try:
+                if from_pt:
                     import torch
 
                     state = torch.load(state_f)
-                    state = {k: v.numpy() for k, v in state.items()}
-                    state = cls.convert_from_pytorch(state, config)
-                    state = unflatten_dict({tuple(k.split(".")[1:]): v for k, v in state.items()})
-                except UnpicklingError:
-                    raise EnvironmentError(
-                        f"Unable to convert model {archive_file} to Flax deserializable object. "
-                        "Supported format are PyTorch archive or Flax msgpack"
-                    )
 
-        return cls(config, state, *model_args, **model_kwargs)
+                    state = convert_state_dict_from_pt(cls, state, config)
+                else:
+                    state = from_bytes(cls, state_f.read())
+            except UnpicklingError:
+                raise EnvironmentError(
+                    f"Unable to convert pytorch model {archive_file} to Flax deserializable object. "
+                )
+
+        # init random models
+        model = cls(config, *model_args, **model_kwargs)
+
+        # if model is base model only use model_prefix key
+        if cls.base_model_prefix not in dict(model.params) and cls.base_model_prefix in state:
+            state = state[cls.base_model_prefix]
 
-    def save_pretrained(self, folder):
-        folder_abs = os.path.abspath(folder)
+        # flatten dicts
+        state_dict = cls.flatten_frozen_state(state)
+        rnd_dict = cls.flatten_frozen_state(model.params)
```

Comment:
What does rnd stand for? `random`? If that's the case, let's go with the full name :-)